cmd/server/main.go

```diff
@@ -170,6 +173,11 @@ func startCRI(ctx context.Context, wg *sync.WaitGroup, config Config) error {
 	if err != nil {
 		return fmt.Errorf("could not start CRI listener: %v ", err)
 	}
+	err = os.Chmod(config.ListenSocket, defaultSocketPerms)
```

Comment:
After checking, you could use `github.com/sylabs/singularity/pkg/util/unix` and call `lis ,err := unix.CreateSocket` which create a socket with 0600 by default. It's up to you

Reply:
done!
thank you for suggestion